react/README.md

```diff
@@ -363,7 +364,192 @@ export default class Candidate extends React.Component {
 
 ## Helper components
 
-Coming soon...
+When a component contains a lot of markup or it contains significant logic that determines how its markup should appear, use helper components to keep `render()` as small as possible. Instead of using `class` declarations for these helper components, use [stateless functions](https://facebook.github.io/react/docs/reusable-components.html#stateless-functions). Because these components are only useful to the main component and only exist to keep `render()` lean, these helper components shouldn't be placed in their own files, nor should they be `export`ed within the main component.
+
+Let's look at a simple example. Let's say you had a global footer that contained a section with links to important top-level pages, another section with a bunch of links for SEO (😐), a section of links to all of  your top-level domains (.com, .co.uk, etc), and a final section of links to all of your social media accounts.
+
+Properly using helper components, this would look like:
+
+```js
+// good (clean render w/ help of helper components)
+
+// using arrow functions for stateless functions
+const SiteLinks = () => (
+    <ul className="global-footer__site-links">
+        <li><a href="/about">About</a></li>
+        <li><a href="/blog">Blog</a></li>
+        <li><a href="/help">Help</a></li>
+        <li><a href="/careers">Careers</a></li>
+    </ul>
+);
+
+// arrow function takes in props object
+const SeoLinks = (props) => {
+    let linkItems = props.links.map((link) => (
+        <li><a key={link.url} href={link.url}>{link.label}</a></li>
+    ));
+
+    return (
+        <ul className="global-footer__seo-links">
+            {linkItems}
+        </ul>
+    );
+};
+
+// arrow function immediately destructures props object into
+// `allTlds` and `currentTld`
+const DomainLinks = ({allTlds, currentTld}) => {
+    // use destructuring to immediately pull out `allTlds` & `currentTld`
+
+    // filter out the current tld and then map to <li>
+    let domainItems = allTlds.filter((tldInfo) => tldInfo.tld !== currentTld)
+        .map(({tld, url, label}) => (
+            <li><a key={tld} href={url}>{label}</a></li>
+        ));
+
+    return (
+        <ul className="global-footer__domain-links">
+            {domainItems}
+        </ul>
+    );
+};
+
+// arrow function immediately destructures props object into `links`
+const SocialLinks = ({links}) => {
+    if (!links) {
```

Comment:
Yeah that's in the gray area because really that's just a little fact about how React works, not really a best practice. But i'll still include the comment 😄